pkg/standalone-utils/contracts/BalancerFeeBurner.sol

```diff
@@ -0,0 +1,124 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.24;
+
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import { IProtocolFeeBurner } from "@balancer-labs/v3-interfaces/contracts/standalone-utils/IProtocolFeeBurner.sol";
+import { IBalancerFeeBurner } from "@balancer-labs/v3-interfaces/contracts/standalone-utils/IBalancerFeeBurner.sol";
+import { IVaultErrors } from "@balancer-labs/v3-interfaces/contracts/vault/IVaultErrors.sol";
+import { IVault } from "@balancer-labs/v3-interfaces/contracts/vault/IVault.sol";
+import "@balancer-labs/v3-interfaces/contracts/vault/VaultTypes.sol";
+
+import {
+    ReentrancyGuardTransient
+} from "@balancer-labs/v3-solidity-utils/contracts/openzeppelin/ReentrancyGuardTransient.sol";
+import { SingletonAuthentication } from "@balancer-labs/v3-vault/contracts/SingletonAuthentication.sol";
+import { VaultGuard } from "@balancer-labs/v3-vault/contracts/VaultGuard.sol";
+
+contract BalancerFeeBurner is IBalancerFeeBurner, ReentrancyGuardTransient, VaultGuard, SingletonAuthentication {
+    using SafeERC20 for IERC20;
+
+    mapping(IERC20 => SwapPathStep[] steps) internal _burnSteps;
+
+    constructor(IVault vault) VaultGuard(vault) SingletonAuthentication(vault) {
+        // solhint-disable-previous-line no-empty-blocks
+    }
+
+    /// @inheritdoc IBalancerFeeBurner
+    function setBurnPath(IERC20 feeToken, SwapPathStep[] calldata steps) external authenticate {
```

Comment:
We want to wire these burners reducing governance whenever possible.

One option would be getting the fee recipient from the fee sweeper (keep a reference to the fee sweeper in this contract) and do `feeSweeperOrGovernance`: if fee recipient is the caller, allow setting the path, else defer auth to governance as usual.

That's just an option; if you think there's a better one I'm also open to it.

Reply:
Sounds like a good approach; would just be this (non-exclusive role, as there would always be a fee recipient):

```
modifier onlyFeeRecipientOrGovernance() {
    address roleAddress = _protocolFeeSweeper.getFeeRecipient();

    _ensureAuthenticatedByRole(address(this), roleAddress);
    _;
}
```